src/tests/functional/plugin/cpu/single_layer_tests/interaction.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/base/ov_subgraph.hpp"
+#include "ie_precision.hpp"
+#include "test_utils/fusing_test_utils.hpp"
+#include "ngraph_functions/builders.hpp"
+#include <common_test_utils/ov_tensor_utils.hpp>
+#include <string>
+#include <tuple>
+
+using namespace CPUTestUtils;
+using namespace ov::test;
+using namespace ngraph;
+
+namespace CPULayerTestsDefinitions {
+using InteractionLayerCPUTestParams = std::tuple<ElementType>;
+
+class IntertactionLayerCPUTest : public testing::WithParamInterface<InteractionLayerCPUTestParams>,
+                            virtual public SubgraphBaseTest,
+                            public CPUTestsBase {
+public:
+    static std::string getTestCaseName(const testing::TestParamInfo<InteractionLayerCPUTestParams>& obj) {
+        ElementType inType;
+        std::tie(inType) = obj.param;
+        std::ostringstream results;
+        results << "Prc=" << inType;
+        return results.str();
+    }
+
+    void generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+        for (int i = 0; i < funcInputs.size(); ++i) {
+            const auto& funcInput = funcInputs[i];
+            ov::Tensor tensor;
+
+            tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i], 15, 0, 32768);
+
+            inputs.insert({funcInput.get_node_shared_ptr(), tensor});
+        }
+    }
+
+protected:
+    void SetUp() override {
+        ElementType inType;
+        std::tie(inType) = GetParam();
+        selectedType = std::string("ref_any_") + InferenceEngine::details::convertPrecision(inType).name();
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+        targetStaticShapes.push_back({Shape{3, 4}});
+        auto dense_feature = std::make_shared<ngraph::opset1::Parameter>(element::f32, PartialShape{3, 4});
+        NodeVector features{dense_feature};
+        std::vector<float> emb_table_value = {-0.2, -0.6, -0.1, -0.4, -1.9, -1.8, -1.,
```

Comment:
Better to move blobs initialization to fn *generate_inputs* to check more than one case. Tests for dynamic shapes are also required.

Reply:
Sure, I remove hard-code shape in c67fb3cc606836770a72b723ec7843f237c2864a